ext/pdo_mysql/pdo_mysql.c

```diff
@@ -68,10 +68,7 @@ static MYSQLND * pdo_mysql_convert_zv_to_mysqlnd(zval * zv)
 			return NULL;
 		}
 
-		if (dbh->driver != &pdo_mysql_driver) {
-			php_error_docref(NULL, E_WARNING, "Provided PDO instance is not using MySQL but %s", dbh->driver->driver_name);
-			return NULL;
-		}
+		ZEND_ASSERT(dbh->driver == &pdo_mysql_driver);
```

Comment:
The above `!dbh` check can be an assertion, but I don't think this one can be. The function accepts a generic zval, so there's no guarantee it uses the mysql driver.

The whole "reverse API" doesn't appear to be actually used inside PHP (looking around, it may be intended for mysqlnd_uh extension). Maybe it can be dropped in the future, but for now I'd leave it alone.